packages/framework/src/DependencyInjection/Compiler/LazyRedisCompilerPass.php

```diff
@@ -24,5 +25,9 @@ public function process(ContainerBuilder $container): void
                 $definition->setLazy(true);
             }
         }
+
+        if ($container->getParameter('kernel.environment') === EnvironmentType::TEST) {
+            $container->getDefinition('snc_redis.test')->setPublic(true);
+        }
```

Comment:
Is any particular reason to have the definition here in a compiler pass?
I would prefer to put it in `services_test.yml` if necessary.

Reply:
Client service is created dynamically and is not possible to specify it in services_test file